src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -345,6 +356,7 @@ def __init__(
         self.best_loss = 0.0
         self.progress_bar_refresh_rate = progress_bar_refresh_rate
         self.progress_bar = None
+        self.should_stop_rank = 0
```

Comment:
I vote for not doing it this way. This makes it complicated where I believe it doesn't need to be. At any point, an arbitrary number of processes can decide to "should_stop=True". The choice here to choose "one of them" is as arbitrary as choosing just rank 0. We should just let rank 0 return the results IMO. See again my suggestion here: https://github.com/Lightning-AI/lightning/pull/15304/files#r1009288887

Reply:
the possibility is that rank 0 might not have good results to generate a relevant lr suggestion.

Reply:
We don't care about that. User needs to "tune" the tuner anyway, as this can happen on a single-device configuration too!